File: examples/export-slide-deck/rest/java/ExportSlideDeck.java
// Exports a Google Slides presentation as a PDF using Browserless.
//
// Run: javac ExportSlideDeck.java && java ExportSlideDeck

import java.net.URI;
import java.net.http.*;
import java.nio.file.*;

public class ExportSlideDeck {
    public static void main(String[] args) throws Exception {
        String token = "YOUR_API_TOKEN_HERE";
        String presentationId = "YOUR_PRESENTATION_ID";
        String endpoint = "https://production-sfo.browserless.io/pdf?token=" + token;

        String payload = "{"
            + "\"url\": \"https://docs.google.com/presentation/d/" + presentationId + "/export/pdf\","
            + "\"options\": {\"printBackground\": true, \"format\": \"A4\", \"landscape\": true}"
            + "}";

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(payload))
            .build();

        HttpResponse<byte[]> response = client.send(request, HttpResponse.BodyHandlers.ofByteArray());
        Files.write(Path.of("slide-deck.pdf"), response.body());
        System.out.println("Exported to slide-deck.pdf");
    }
}

File: examples/export-slide-deck/rest/python/export_slide_deck.py
# Exports a Google Slides presentation as a PDF using Browserless.
#
# Install: pip install requests
# Run:     python export_slide_deck.py

import requests

PRESENTATION_ID = 'YOUR_PRESENTATION_ID'
TOKEN = 'YOUR_API_TOKEN_HERE'

response = requests.post(
    f'https://production-sfo.browserless.io/pdf?token={TOKEN}',
    json={
        'url': f'https://docs.google.com/presentation/d/{PRESENTATION_ID}/export/pdf',
        'options': {
            'printBackground': True,
            'format': 'A4',
            'landscape': True,
        },
    },
)

with open('slide-deck.pdf', 'wb') as f:
    f.write(response.content)

print('Exported to slide-deck.pdf')
